### PR TITLE
vk-bootstrap: init at 0.5

### LINKED ISSUE #183795: Request: vk-bootstrap vulkan library
**Project description**
A utility library that jump starts initialization of Vulkan. Helps writing simple boilerplate when dealing with Vulkan projects, helping beginners get into the ecosystem.

**Metadata**

* homepage URL: https://github.com/charles-lunarg/vk-bootstrap (and possibly https://vkguide.dev/)
* source URL: https://github.com/charles-lunarg/vk-bootstrap
* license: mit
* platforms: all (I think)

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Description of changes
Fixes https://github.com/NixOS/nixpkgs/issues/183795

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/vk-bootstrap/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  };
+
+  postPatch = ''
+    # Upstream uses cmake FetchContent to resolve glfw and catch2
+    # needed for examples and tests
+    sed -iE 's=add_subdirectory(ext)==g' CMakeLists.txt
+    sed -iE 's=Catch2==g' tests/CMakeLists.txt
+  '';
+
+  nativeBuildInputs = [ cmake ];
```
Comment (fufexan) [inline pkgs/development/libraries/vk-bootstrap/default.nix:28]: I think these generally come after `src`, then the phases follow.

File: pkgs/development/libraries/vk-bootstrap/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+, fetchFromGitHub
+, cmake
+, vulkan-headers
+, glfw
+, catch2
+}:
+
+stdenv.mkDerivation rec {
+  pname = "vk-bootstrap";
+  version = "0.5";
```
Comment (pbsds) [inline pkgs/development/libraries/vk-bootstrap/default.nix:12]:
```suggestion
  version = "0.5";
  outputs = [ "out" "dev" ];
```

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 184600` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vk-bootstrap</li>
  </ul>
</details>

**Comment by Ashe** [comment]
Are there any updates on this? :smile:

**Review by pbsds — APPROVED** [review]
Lgtm syntax wise

File: pkgs/development/libraries/vk-bootstrap/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ vulkan-headers glfw catch2 ];
+
+  cmakeFlags = [
+    "-DVK_BOOTSTRAP_VULKAN_HEADER_DIR=${vulkan-headers}/include"
+  ];
+
+  meta = with lib; {
+    description = "Vulkan Bootstrapping Iibrary";
```
Comment (viraptor) [inline pkgs/development/libraries/vk-bootstrap/default.nix:37]:
```suggestion
    description = "Vulkan Bootstrapping Library";
```

**Comment by Ashe** [comment]
Thanks for the review, I'm still hoping we can get this submitted soon! :)

**Pull request merged**